File: Arrays & Hashing/Java/validAnagrams.java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false; 
        }

        HashMap<Character, Integer> count_s = new HashMap<Character, Integer>(); 
        HashMap<Character, Integer> count_t = new HashMap<Character, Integer>();

        for (int i = 0; i < s.length(); i++) {
            char char_i = s.charAt(i);
            count_s.computeIfPresent(char_i, (k, v) -> v + 1);
            count_s.putIfAbsent(char_i, 1);
        }

        for (int j = 0; j < t.length(); j++) {
            char char_j = s.charAt(j);
            count_t.computeIfPresent(char_j, (k, v) -> v + 1); 
            count_t.putIfAbsent(char_j, 1); 
        } 

        if (count_s.equals(count_t)) {
            return true;
        }
        else {
            return false; 
        }
    }
}

File: Arrays & Hashing/python/valid_anagrams.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): 
            return False 

        count_s, count_t = {}, {}

        # for i in range(len(s)): 
        #     if s[i] not in count_s: 
        #         count_s[s[i]] = 1
        #     else: 
        #         count_s[s[i]] += 1

        #     if t[i] not in count_t: 
        #         count_t[t[i]] = 1
        #     else: 
        #         count_t[t[i]] += 1

        for i in range(len(s)):
            count_s[s[i]] = count_s.get(s[i], 0) + 1
            count_t[t[i]] = count_t.get(t[i], 0) + 1
                        
        return count_s == count_t